flake.nix

```diff
@@ -351,27 +401,101 @@
               autoPatchelf ./.venv
             '';
             preShellHook = ''
-              sed -i "s|solc = .*|solc = \"${solcDefault}/bin/solc\"|g" ethereum/contracts/foundry.toml
+              sed "s|# solc = .*|solc = \"${solcDefault}/bin/solc\"|g" \
+                ethereum/contracts/foundry.toml.in > \
+                ethereum/contracts/foundry.toml
+            '';
+            postShellHook = ''
+              ${pre-commit-check.shellHook}
             '';
           };
           defaultDevShell = buildDevShell [ ];
           smoketestsDevShell = buildDevShell [ hoprd hopli ];
+          update-github-labels = flake-utils.lib.mkApp {
+            drv = pkgs.writeShellScriptBin "update-github-labels" ''
+              set -eu
+              # remove existing crate entries (to remove old crates)
+              yq 'with_entries(select(.key != "crate:*"))' .github/labeler.yml > labeler.yml.new
+              # add new crate entries for known crates
+              for f in `find . -mindepth 2 -name "Cargo.toml" -type f ! -path "./vendor/*"`; do
+              	env \
+              		name="crate:`yq '.package.name' $f`" \
+              		dir="`dirname $f`/**" \
+              		yq -n '.[strenv(name)][0]."changed-files"[0]."any-glob-to-any-file"[0] = env(dir)' >> labeler.yml.new
+              done
+              mv labeler.yml.new .github/labeler.yml
+            '';
+          };
```

Comment:
Can we also delete obsoleted labels in GH? Because this can easily explode with renaming.

Reply:
Which ones are you referring to?

Reply:
When you create a new crate, a label is created in github... then when you rename the crate, that label lives on... should we not find a way to delete them?

Reply:
I see. Let's tackle this separately.